src/menus/menus.service.ts

```diff
@@ -106,6 +106,17 @@ export class MenusService {
         },
       },
     },
+    {
+      $addFields: {
+        menus: {
+          $filter: {
+            input: '$menus',
+            as: 'menu',
+            cond: { $ne: ['$$menu', null] },
+          },
+        },
+      },
+    },
```

Comment:
This line is use to remove the null menu inside the array menu when querying with status